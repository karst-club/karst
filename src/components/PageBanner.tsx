import React from 'react';
import styled from 'styled-components';
import bannerImage from '../images/ship_scene.png';

const PageHeaderImage = styled.img`
  display: block;
  object-fit: cover;
  border-radius: 0px;
  width: 100%;
  height: 30vh;
  opacity: 1;
  object-position: center 39.59%;
`;

const PageHeaderWrapper = styled.div`
  position: relative;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 30vh;
  cursor: default;
  background-color: #282c34;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

const PageBanner: React.FC = () => {
  return (
    <header>
      <PageHeaderWrapper>
        <PageHeaderImage src={bannerImage} alt="header" />
      </PageHeaderWrapper>
    </header>
  );
};

export default PageBanner;
