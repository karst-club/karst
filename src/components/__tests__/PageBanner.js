import React from 'react';
import renderer from 'react-test-renderer';
import PageBanner from '../PageBanner';

describe('PageBanner', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PageBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
