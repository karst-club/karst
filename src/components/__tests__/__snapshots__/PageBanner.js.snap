// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBanner renders correctly 1`] = `
<header>
  <div
    className="sc-gsTCUz inYfpR"
  >
    <img
      alt="header"
      className="sc-bdfBwQ jXVdju"
      src="test-file-stub"
    />
  </div>
</header>
`;
